src/prototype.rs

```diff
@@ -72,20 +80,83 @@ pub trait Prototypical: 'static + Send + Sync {
 	) -> EntityCommands<'a, 'b> {
 		let mut entity = commands.spawn();
 		let mut proto_commands = self.create_commands(entity, data);
-		for component in self.iter_components() {
-			component.insert_self(&mut proto_commands, asset_server);
-		}
+
+		spawn_internal(
+			self.name(),
+			self.template(),
+			self.iter_components(),
+			&mut proto_commands,
+			data,
+			asset_server,
+			&mut IndexSet::default(),
+		);
 
 		proto_commands.into()
 	}
 }
 
+/// Internal method used for recursing up the template hierarchy and spawning components
+/// from the top to the bottom
+fn spawn_internal<'a>(
+	name: &'a str,
+	template: Option<&'a str>,
+	components: Iter<'a, Box<dyn ProtoComponent>>,
+	proto_commands: &mut ProtoCommands,
+	data: &'a Res<ProtoData>,
+	asset_server: &Res<AssetServer>,
+	traversed: &mut IndexSet<&'a str>,
+) {
+	// We insert first on the off chance that someone made a prototype its own template...
+	traversed.insert(name);
+
+	match template {
+		Some(template_name) if traversed.contains(template_name) => {
+			// ! === Found Circular Dependency === ! //
+			let tree: String = traversed
+				.iter()
+				.map(|n| format!("'{}' -> ", n))
+				.collect::<String>()
+				.add(&format!("'{}'", template_name));
+			warn!(
+				"{} ({})\n\t{}",
+				"Found a circular dependency when trying to spawn a prototype!",
+				tree,
+				"The rest of the spawn has been skipped, but make sure you remove any template that might call itself!"
+			);
+		}
+		Some(template_name) => {
+			// === Spawn Template === //
+			if let Some(parent) = data.get_prototype(template_name) {
+				spawn_internal(
+					parent.name(),
+					parent.template(),
+					parent.iter_components(),
+					proto_commands,
+					data,
+					asset_server,
+					traversed,
+				);
+			}
+		}
+		_ => (),
+	}
+
+	// === Spawn Self === //
+	for component in components {
+		component.insert_self(proto_commands, asset_server);
+	}
+}
+
 /// The default prototype object, providing the basics for the prototype system
 #[derive(Serialize, Deserialize)]
 pub struct Prototype {
 	/// The name of this prototype
 	pub name: String,
+	/// The name of this prototype's template (if any)
+	#[serde(default)]
+	pub template: Option<String>,
```

Comment:
Would it be a significant change in scope to make multiple templating an option?

Reply:
A concern there is it means we might also have to deal with multiple inheritance issues. Such as if two templates have the same components defined, which one gets implemented? We could do further checks to warn about this, but I think any solution would be nondeterministic and/or cause confusion.

One simple solution, if we really wanted it, could be to allow multiple ancestors, but force overrides in decreasing order. For example:
```yaml
templates:
  - Template A # Components always retained
  - Template B # Conflicting components overridden by A
  - Template C # Conflicting components overridden by A & B 
```

But I'm not sure if that's a good idea or not. Thoughts?

Reply:
I've put this logic in its own PR here: #3